### PR TITLE
docs: Getting started sandbox

### PR DESCRIPTION
**Pull request by Jayclifford345**
This PR adds the metadata required to sandbox the getting started guide of tempo. This will allow users to run the tutorial in an online VM. To test you can use this developer version: https://killercoda.com/grafana-labs-chaos-testing/course/tempo/quick-start

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/getting-started/docker-example.md [hunk 1]
```diff
@@ -58,16 +91,51 @@ To follow this guide, you need:
...
+3. To stop the services:
+    ```bash
     docker compose down -v
     ```
     
+<!-- INTERACTIVE page step2.md END -->
+
+<!-- INTERACTIVE page finish.md START -->
+
+# Next steps
```
[the remark this answers is not in the document]
Comment (knylander-grafana) [reply docs/sources/tempo/getting-started/docker-example.md:122]: This is an automated thing from the validator. I'm not sure how to escape around this. 

@jdbaldry Have you dealt with this before?

**Comment by knylander-grafana** [comment]
@Jayclifford345 This is SO cool! I love having this done.

File: docs/sources/tempo/getting-started/docker-example.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdbaldry) [reply docs/sources/tempo/getting-started/docker-example.md:122]: Ooh this is a good one. This is usually a correct positive because a page shouldn't have two H1 headings but it makes sense use a H1 here because it's the first heading in that course's page.

I think `doc-validator` needs a comment based mechanism for ignoring parts of the code and I may be able to find some time to work on that today.

In the short term, you can make this section start with an H2 and know that the transformer tool will make it an H1: https://github.com/grafana/killercoda/blob/staging/tools/transformer/transform.go#L233-L262

[the remark this answers is not in the document]
Comment (Jayclifford345) [reply docs/sources/tempo/getting-started/docker-example.md]: Hi all, sorry that was completely my mistake. It should have been a H2 so now made that fix!

**Comment by Jayclifford345** [comment]
Hahahha now worries @knylander-grafana! It was really fun, I am just working through a PR with @jdbaldry within the transformer and once we merge that I can this green for your review :)

**Comment by Jayclifford345** [comment]
Okay @knylander-grafana, we are ready for your review!

File: docs/sources/tempo/getting-started/docker-example.md [hunk 2]
```diff
@@ -16,12 +34,7 @@ The Tempo examples running with Docker using docker-compose include a version of
 This quick start guide uses the `local` example running Tempo as a single binary (monolithic). Any data is stored locally in the `tempo-data` folder.
 To learn more, read the [local storage example README](https://github.com/grafana/tempo/blob/main/example/docker-compose/local).
 
-## Alternative: Complete MLTP example
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:19]: Why did you remove this section?

**Comment by knylander-grafana** [comment]
> Okay @knylander-grafana, we are ready for your review!

What is the best way for me to review this PR? Are there things I should test for the automated content updates or just check out and build locally?

**Comment by jdbaldry** [comment]
> > Okay @knylander-grafana, we are ready for your review!
> 
> @Jayclifford345 What is the best way for me to review this PR? Are there things I should test for the automated content updates or just check out and build locally?

You primarily want to check that none of the INTERACTIVE directives have damaged the rendered documentation page. I think Jay will check on his side that the transformed tutorial looks good.

**Comment by jdbaldry** [comment]
Also, it would be great to have a technical writer review any of the instructional content even if it doesn't end up in the documentation page so that it conforms to our style and is clear for the users of the course.

File: docs/sources/tempo/getting-started/docker-example.md [hunk 3]
```diff
@@ -31,6 +44,26 @@ To follow this guide, you need:
...
+It's a fully configured environment with all the dependencies already installed.
+
+![Interactive](/media/docs/tempo/tempo-ile.svg)
+
+Provide feedback, report bugs, and raise issues in the [Grafana Killercoda repository](https://github.com/grafana/killercoda).
+{{< /admonition >}}
+
+<!-- INTERACTIVE ignore END -->
+
+{{<docs/ignore>}}
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:59]: Normally we use and admonition shortcode. Do we need to remove any shortcodes for Killercode?

File: docs/sources/tempo/getting-started/docker-example.md [hunk 4]
```diff
@@ -31,6 +44,26 @@ To follow this guide, you need:
...
+> **Note**: A setup script is running in the background to install the necessary dependencies. This should take no longer than 30 seconds. Your instance will be ready to use once you `Setup complete. You may now begin the tutorial`.
+{{</docs/ignore>}}
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:61]: Normally we use and admonition shortcode. Do we need to remove any shortcodes when we use markdown for Killercoda and Hugo?

Comment (jdbaldry) [reply docs/sources/tempo/getting-started/docker-example.md:61]: We have support for some shortcodes in Killercoda but not all of them. Admonitions are not yet supported so we write Killercoda notes in plain Markdown. I've created the documentation in https://github.com/grafana/killercoda/pull/118.

Comment (jdbaldry) [reply docs/sources/tempo/getting-started/docker-example.md:61]: https://github.com/grafana/killercoda/pull/121 🎉

Comment (Jayclifford345) [reply docs/sources/tempo/getting-started/docker-example.md]: tested and works perfectly: https://killercoda.com/grafana-dev-testing/course/tempo/quick-start

File: docs/sources/tempo/getting-started/docker-example.md [hunk 5]
```diff
@@ -31,6 +44,28 @@ To follow this guide, you need:
 - Docker
 - The docker-compose plugin
 
+{{< admonition type="tip" >}}
+Alternatively, you can try out this example in our interactive learning environment: [Quick start for Tempo](https://killercoda.com/grafana-labs/course/tempo/quick-start).
+
+It's a fully configured environment with all the dependencies already installed.
+
+![Interactive](/media/docs/tempo/tempo-ile.svg)
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:52]: This large banner is out of place in the docs. The admonition already highlights the tutorial. Isn't the text in the green highlight enough? 
```suggestion
```
This is what it looks like: 
image

**Review by knylander-grafana — CHANGES_REQUESTED** [review]
Thank you for converting this quick start to a Killer Coda example! Can't wait to see what the community thinks about the interactive example. :) 

I used this review to take the opportunity to correct some issues when I wrote this and hopefully improve on some of the content. I've updated screenshots and moved some content around slightly. 

The overall display in the local build didn't have any obvious errors. I'm glad I did build it locally because otherwise I wouldn't have seen the out-of-date screenshots.  I'll remove the local screenshots that are used in this doc.

**Comment by knylander-grafana** [comment]
This PR removes the out of date images from the other documentation and deletes them from the repository: https://github.com/grafana/tempo/pull/4066

File: docs/sources/tempo/getting-started/docker-example.md [hunk 6]
```diff
@@ -58,16 +99,52 @@ To follow this guide, you need:
...
+<!-- INTERACTIVE page finish.md START -->
+
+## Next steps
+
+You have successfully set up Tempo and Grafana to explore traces generated by the k6-tracing service.
+
+### Alternative: Complete MLTP example
+
+If you would like to use a demo with Tempo, Mimir, Loki, and Grafana, then try the [Introduction to Metrics, Logs, Traces, and Profiling in Grafana](https://github.com/grafana/intro-to-mlt).
+`Intro-to-mltp` provides a self-contained environment for learning about Mimir, Loki, Tempo, and Grafana.
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:138]:
```suggestion
`Intro-to-mltp` provides a self-contained environment for learning about Mimir, Loki, Tempo, Pyroscope, and Grafana.
```

File: docs/sources/tempo/getting-started/docker-example.md [hunk 7]
```diff
@@ -58,16 +99,52 @@ To follow this guide, you need:
...
+
+<!-- INTERACTIVE page finish.md START -->
+
+## Next steps
+
+You have successfully set up Tempo and Grafana to explore traces generated by the k6-tracing service.
+
+### Alternative: Complete MLTP example
+
+If you would like to use a demo with Tempo, Mimir, Loki, and Grafana, then try the [Introduction to Metrics, Logs, Traces, and Profiling in Grafana](https://github.com/grafana/intro-to-mlt).
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:137]:
```suggestion
If you would like to use a demo with multiple telemetry signals, then try the [Introduction to Metrics, Logs, Traces, and Profiling in Grafana](https://github.com/grafana/intro-to-mlt).
```

File: docs/sources/tempo/getting-started/docker-example.md [hunk 8]
```diff
@@ -3,8 +3,26 @@ title: Quick start for Tempo
...
+  title: Quick start for Tempo
+  description: Use Docker to quickly view traces using K-6 and Tempo
+  details:
+      intro:
+         foreground: docker-compose-update.sh
+  preprocessing:
+    substitutions:
+      - regexp: ../../
+        replacement: /docs/tempo/latest/
+      - regexp: '{{< relref "./instrumentation" >}}'
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:16]: @jdbaldry I know the Witers' Toolkit recommends using full URLs. Should I update the URls to all use the full URLs or does it matter that much?

File: docs/sources/tempo/getting-started/docker-example.md [hunk 9]
```diff
@@ -58,16 +99,52 @@ To follow this guide, you need:
...
+
+If you would like to use a demo with multiple telemetry signals, then try the [Introduction to Metrics, Logs, Traces, and Profiling in Grafana](https://github.com/grafana/intro-to-mlt).
+`Intro-to-mltp` provides a self-contained environment for learning about Mimir, Loki, Tempo, Pyroscope, and Grafana.
+The project includes detailed explanations of each component and annotated configurations for a single-instance deployment.
+Data from `intro-to-mltp` can also be pushed to Grafana Cloud.
+
+### Further reading
+Here are some resources to help you learn more about Tempo:
+* [Instrumenting your application]({{< relref "./instrumentation" >}})
+* [Setup guides for Tempo]({{< relref "../setup" >}})
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:145]:
```suggestion
* [Instrumenting your application](https://grafana.com/docs/tempo/<TEMPO_VERSION>/getting-started/instrumentation/)
* [Setup guides for Tempo](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/)
```

**Comment by knylander-grafana** [comment]
I've resolved most of the issues in the PR. Per Jack's suggestion, I've asked for another writer to review the PR.

File: docs/sources/tempo/getting-started/docker-example.md (hunk 8 above)
Comment (jdbaldry) [reply docs/sources/tempo/getting-started/docker-example.md:16]: Using full URLs would let us remove these preprocessing statements which would be nice. The transformer tool knows how to convert full URLs with version substitution into Killercoda URLs.

**Comment by jdbaldry** [comment]
> I've resolved most of the issues in the PR. Per Jack's suggestion, I've asked for another writer to review the PR. I'll clear the request changes after the other PR review.

Oh I didn't mean another technical writer necessarily but I do think peer review is a good way to share knowledge and practice :)

File: docs/sources/tempo/getting-started/docker-example.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jayclifford345) [reply docs/sources/tempo/getting-started/docker-example.md:50]: Hi @knylander-grafana, 
Not a problem we can get that removed. We have made it standard across the other pages. Agreed it might be to big. I can remove for now and speak to the design team to see if we can make something less aggressive but still indicate that is page is part of the greater interactive sandbox effort

**Comment by Jayclifford345** [comment]
Hi all, I have removed the image for the Interactive learning environment. I agree with Kim and I am now in an artistic crisis over my own banner design :D. So, I will drop a note to the design team to see if we can get something less intrusive created but still indicate these sandboxes are part of the larger interactive learning effort

**Comment by Jayclifford345** [comment]
I think for now however we can go without the image. So happy for it to be merged when ready

**Comment by knylander-grafana** [comment]
> Hi all, I have removed the image for the Interactive learning environment. I agree with Kim and I am now in an artistic crisis over my own banner design :D. So, I will drop a note to the design team to see if we can get something less intrusive created but still indicate these sandboxes are part of the larger interactive learning effort

I lke the idea of the banner. The current design is a big overwhelming on the page. Do you have banners on the other Killer coda tutorial doc pages within the documentation or only on the Killer Coda published sites?

File: docs/sources/tempo/getting-started/docker-example.md (hunk 8 above)
Comment (knylander-grafana) [reply docs/sources/tempo/getting-started/docker-example.md:16]: Okay. If I change the URLs in the body, then the regexp can be removed?

File: docs/sources/tempo/getting-started/docker-example.md [hunk 10]
```diff
@@ -58,16 +98,52 @@ To follow this guide, you need:
...
    ```
 
-1. As part of the docker compose manifest, Grafana is now accessible on port 3000. Open a browser to localhost:3000 and sign in to Grafana.
+<!-- INTERACTIVE page step1.md END -->
+
+<!-- INTERACTIVE page step2.md START -->
+
+## Explore the traces in Grafana
+
+As part of the docker compose manifest, Grafana is now accessible on port 3000.
```
Comment (clayton-cornell) [inline docs/sources/tempo/getting-started/docker-example.md:107]:
```suggestion
As part of the Docker Compose manifest, Grafana is now accessible on port 3000. 
```

According to:  https://docs.docker.com/compose/  it should be initial caps when describing the command.

File: docs/sources/tempo/getting-started/docker-example.md [hunk 11]
```diff
@@ -13,26 +31,43 @@ Every example has a `docker-compose.yaml` manifest that includes all of the opti
...
+- [Docker](https://docs.docker.com/compose/install/)
+- The Docker Compose plugin (included with Docker Desktop)
+
+{{< admonition type="tip" >}}
+Alternatively, you can try out this example in our interactive learning environment: [Quick start for Tempo](https://killercoda.com/grafana-labs/course/tempo/quick-start).
+
+It's a fully configured environment with all the dependencies installed.
+
+
+Provide feedback, report bugs, and raise issues in the [Grafana Killercoda repository](https://github.com/grafana/killercoda).
```
Comment (clayton-cornell) [inline docs/sources/tempo/getting-started/docker-example.md:51]:
```suggestion

Provide feedback, report bugs, and raise issues in the [Grafana Killercoda repository](https://github.com/grafana/killercoda).
```

Extra CR/LF

Alloy includes a banner image at this point. There's no banner here?

Also, this is new compared to the Alloy version. Should I include this over in the Alloy tutorials?

File: docs/sources/tempo/getting-started/docker-example.md (hunk 8 above)
Comment (knylander-grafana) [reply docs/sources/tempo/getting-started/docker-example.md:16]: Okay. I'll remove the preprocess notes from the frontmatter.

File: docs/sources/tempo/getting-started/docker-example.md (hunk 11 above)
Comment (knylander-grafana) [reply docs/sources/tempo/getting-started/docker-example.md:51]: @Jayclifford345 What do you think?

File: docs/sources/tempo/getting-started/docker-example.md (hunk 10 above)
Comment (knylander-grafana) [reply docs/sources/tempo/getting-started/docker-example.md:107]: Good catch. I found this earlier and corrected it but missed this occurrence.

**Review by knylander-grafana — APPROVED** [review]
I think we're good now, Jay. What do you think? Please check Clayton's questions and the frontmatter. In my comments, I changed the relrefs to full URLs, so the preprocessing shouldn't be needed.

File: docs/sources/tempo/getting-started/docker-example.md [hunk 12]
```diff
@@ -3,8 +3,18 @@ title: Quick start for Tempo
 menuTitle: Quick start for Tempo
 description: Use Docker to quickly view traces using K-6 and Tempo
 weight: 600
+killercoda:
+  title: Quick start for Tempo
+  description: Use Docker to quickly view traces using K-6 and Tempo
+  details:
+      intro:
+         foreground: docker-compose-update.sh
+    backend:
```
Comment (knylander-grafana) [inline docs/sources/tempo/getting-started/docker-example.md:12]:
```suggestion
  backend:
    backend:
```

File: docs/sources/tempo/getting-started/docker-example.md (hunk 11 above)
Comment (Jayclifford345) [reply docs/sources/tempo/getting-started/docker-example.md]: Yeah, more than happy to have the banner removed and this section updated. I still think some form of icon or branding, later on, to differentiate it from a normal admonition makes users know this tutorial includes a sandbox from the initial glance.

**Comment by Jayclifford345** [comment]
Hi @knylander-grafana!
Looks great to me, happy to go ahead with the merge. Just give me a ping and I can get the sandbox merged into production on the Killercoda side as well and added to the automation pipeline.

Just FYI it looks like all of @clayton-cornell changes have been added as it looked already updated but apologies if I missed those.

Huge thanks all for reviewing and getting this in. Its a big win!

**Comment by knylander-grafana** [comment]
> Hi @knylander-grafana! Looks great to me, happy to go ahead with the merge. Just give me a ping and I can get the sandbox merged into production on the Killercoda side as well and added to the automation pipeline.
> 
> Just FYI it looks like all of @clayton-cornell changes have been added as it looked already updated but apologies if I missed those.
> 
> Huge thanks all for reviewing and getting this in. Its a big win!

This is awesome! Let's go ahead and merge this. When you get the Killercoda side ready, then we'll merge the backport.
